### ISSUE TITLE
Octave_kernel on Juypyter MacOs

### ISSUE THREAD

**Issue by bitMaestro** [Calysto/octave_kernel#116]
Greetings, I'm attempting to get Octave up and running on Jupyter Notebook. I see the kernel in the drop down menu but then I have an error message:

OSError: Octave Executable not found, please add to path or set"OCTAVE_EXECUTABLE" environment variable....

I've tried all the suggestions and I usually find a solution before relying on help but this time its been about 5 hours and still in trouble. Can you please walk me through this because I'm having technical difficulties.

Thank you in advance.
Tarique

**Comment by blink1073** [comment]
Hi @bitMaestro, if you start a notebook with python kernel and type the following, what is the output?

```
!which octave; echo $OCTAVE_EXECUTABLE
```

**Comment by bitMaestro** [comment]
Hi, I cannot run,

!which octave; echo $OCTAVE_EXECUTABLE

directly from within Jupyter notebook.  I receive an error message, pease
see error message from terminal.

OSError: Octave Executable not found, please add to path or
set"OCTAVE_EXECUTABLE" environment variable

[W 12:35:37.104 NotebookApp] KernelRestarter: restart failed

[W 12:35:37.105 NotebookApp] Kernel 20c82878-d981-4d96-98f1-672feae0d4e6
died, removing from map.

ERROR:root:kernel 20c82878-d981-4d96-98f1-672feae0d4e6 restarted failed!

[W 12:35:37.117 NotebookApp] Kernel deleted before session

[W 12:35:37.118 NotebookApp] 410 DELETE
/api/sessions/a4a1b66c-fd48-4cd3-9f08-f34c11a00863 (::1) 3.24ms referer=
http://localhost:8888/notebooks/Untitled.ipynb


However running the code in the terminal give me this:

$ !which octave; echo $OCTAVE_EXECUTABLE

which python octave; echo $OCTAVE_EXECUTABLE octave; echo $OCTAVE_EXECUTABLE

/Users/tarique/anaconda3/bin/python

/Users/tarique/anaconda3/lib/python3.6/site-packages/octave_kernel octave

/Users/tarique/anaconda3/lib/python3.6/site-packages/octave_kernel

Thanks in advance for your help.
[image: photo]
*Tarique Hasheem*

510-299-2680

<http://www.linkedin.com/in/tariqueh>
Create your own email signature
<https://wisestamp.com/email-install?utm_source=promotion&utm_medium=signature&utm_campaign=create_your_own>

**Comment by blink1073** [comment]
Are you able to start a notebook with a Python kernel?  The above command is intended for a Python kernel.

**Comment by bitMaestro** [comment]
Yes.  If I select Python from the drop down inside of Jupyter notebook
Python will run as it should.

**Comment by blink1073** [comment]
After starting the Python notebook, what is the output of running a cell with the following?  `!which octave; echo $OCTAVE_EXECUTABLE`?

**Comment by ksk8863** [comment]
I have same problem
!which octave; echo $OCTAVE_EXECUTABLE?
The output of running a cell is Object `OCTAVE_EXECUTABLE` not found.

**Comment by asavasci** [comment]
hello, I am using macos, my octave installation is up and running but I cannot use jupyter notebook. When I attempt to create an octave notebook I am getting error below
OSError: Octave Executable not found, please add to path or set"OCTAVE_EXECUTABLE" environment variable

I set an environment variable as 
export PATH="/Applications/Octave.app/Contents/Resources/usr/Cellar/octave-octave-app_4.4.0/4.4.0/bin/octave-cli:$PATH"

into the bash_profile. However, it seems not working. 

Could you advise me how to solve this issue?

**Comment by dsblank** [comment]
Don't put the executable in the path (remove octave-cli)

**Comment by asavasci** [comment]
I tried but getting same error. Should I use different command to set an environment variable?

**Comment by blink1073** [comment]
@asavasci, are you able to run `octave-cli` from a terminal after setting up your PATH as above?

**Comment by rucarden** [comment]
When I installed octave 4.4 using the app bundle at https://wiki.octave.org/Octave_for_macOS, it didn't come with an octave-cli only an octave command.
I just made a link with that name and put in my home bin directory which is in my path.
`ln -s /usr/local/bin/octave ~/bin/octave-cli`
